app/src/androidTest/java/ch/hikemate/app/ui/map/HikeDetailsScreenTest.kt

```diff
@@ -412,14 +432,35 @@ class HikeDetailScreenTest {
           detailedRoute = detailedRoute,
           savedHikesViewModel = savedHikesViewModel,
           emptyList(),
-          HikingLevel.BEGINNER)
+          HikingLevel.BEGINNER,
+          {})
     }
 
     val minuteString =
         String.format(Locale.getDefault(), "%02d", (detailedRoute.estimatedTime % 60).roundToInt())
 
     composeTestRule
-        .onAllNodesWithTag(TEST_TAG_DETAIL_ROW_VALUE)
+        .onAllNodesWithTag(HIKE_DETAILS_TEST_TAG_DETAIL_ROW_VALUE)
         .assertAny(hasText("${minuteString}min"))
   }
+
+  @Test
+  fun hikeDetails_showsRunThisHikeButton_andTriggersOnRunThisHike() {
+    val onRunThisHike = mock<() -> Unit>()
+    composeTestRule.setContent {
+      HikeDetails(
+          detailedRoute = detailedRoute,
+          savedHikesViewModel = savedHikesViewModel,
+          emptyList(),
+          HikingLevel.BEGINNER,
+          { onRunThisHike() })
+    }
+
+    composeTestRule
+        .onNodeWithTag(HIKE_DETAILS_TEST_TAG_RUN_HIKE_BUTTON)
+        .assertIsDisplayed()
+        .performClick()
+
+    verify(onRunThisHike).invoke()
+  }
```

Comment:
Suggestion: I think you pass the action as a parameter because of this test. I think it's okay to test that you call a `navigateTo` in here. It won't work because you don't have a navigation here but you can mock it. (check other tests)

Reply:
Yep!